client/main.go

```diff
@@ -278,6 +280,13 @@ func receive(ctx *cli.Context) error {
 }
 
 func claim(ctx *cli.Context) error {
+	if err := arkSdkClient.Unlock(
```

Comment:
wouldnt be more correct to check is IsUnlocked before trying to unlock an already unlocked SDK? to avoid extra roundtrip

Reply:
This is the only way, AFAIK, that SDK loads primary key from memory, in the [example](https://github.com/sekulicd/ark/blob/52b21c5e687111250b365e89438503e2327ce107/pkg/client-sdk/example/covenant/alice_to_bob.go#L34) itself similar flow is used.

Reply:
The SDK exposes IsLocked https://github.com/sekulicd/ark/blob/52b21c5e687111250b365e89438503e2327ce107/pkg/client-sdk/wallet/wallet.go#L21

Reply:
Yeah, but that wont help, when u init arksdk, pk is not loaded by default, and only place where pk is set is in [Unlock](https://github.com/sekulicd/ark/blob/52b21c5e687111250b365e89438503e2327ce107/pkg/client-sdk/wallet/singlekey/wallet.go#L112) method.
Calling IsLocked is meaningless in this case.